client/src/pages/OnBoard.js

```diff
@@ -0,0 +1,223 @@
+    "PHP",
+    "GoLang",
+    "C#",
+    "C++",
+    "Ruby",
+  ];
+  const levels = ["Beginner", "Advanced", "Expert"];
+
+  const [skillList, setSkillList] = useState([{ language: "", level: "" }]);
+  const [submitClicked, setSubmitClicked] = useState(false);
+
+  const handleInputChange = (event, index) => {
+    const name = event.target.name;
+    const language = event.target.value;
+    const list = [...skillList];
+
+    list[index][name] = language;
+    setSkillList(list);
+  };
+
+  const handleAddClick = (event) => {
+    event.preventDefault();
+    setSkillList([...skillList, { language: "", level: "" }]);
+    setSubmitClicked(false);
+  };
+
+  const handleRemoveClick = (index) => {
+    const list = [...skillList];
+    list.splice(index, 1);
+    setSkillList(list);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    let allValid = true;
+
+    for (let i = 0; i < skillList.length; i++) {
+      allValid =
+        skillList[i].language && skillList[i].level && allValid ? true : false;
+    }
+    if (!allValid) {
+      setSubmitClicked(true);
+    } else {
+      //Send req to API to save skills
+      console.log(skillList);
+      console.log("Submit");
+    }
+  };
+
+  const InputLine = (props) => {
+    return (
+      <Grid container alignItems="center" justify="flex-end">
+        <Grid item xs={12} sm={1} md={1}>
+          <IconButton
+            className={classes.removeIcon}
+            component="span"
+            disabled={
+              props.index === 0 && skillList.length === 1 ? true : false
+            }
+            onClick={() => handleRemoveClick(props.index)}
+          >
+            <RemoveCircleIcon fontSize="large" />
+          </IconButton>
+        </Grid>
+        <Grid item xs={5} sm md>
+          <Typography component="h1" variant="h6">
+            Language:
+          </Typography>
+        </Grid>
+        <Grid item xs={7} sm={4} md={4}>
+          <TextField
+            color="secondary"
+            fullWidth
+            select
+            margin="dense"
+            name="language"
+            onChange={(e) => handleInputChange(e, props.index)}
+            value={props.item.language}
+            label={
+              submitClicked && props.item.language === ""
+                ? "Specify language"
+                : null
+            }
+            variant="outlined"
+            error={submitClicked && props.item.language === "" ? true : false}
+          >
+            <MenuItem value={props.item.language}>
+              {props.item.language}
+            </MenuItem>
+            {languages.map((item, index) => {
+              if (skillList.map((item) => item.language).includes(item)) {
+                return null;
+              } else {
+                return (
+                  <MenuItem key={index} value={item}>
+                    {item}
+                  </MenuItem>
+                );
+              }
+            })}
```

Comment:
I think a cleaner way to write this would be below. Can use filter on the languages and find if that language is in skillList already. I didn't check if this work but the logic is there. Unfortunately this is a nested for loop (O(n^2)), but I don't care since the list of languages are small


```
            {languages.filter((lang) => (
                   skillList.find((skill) => (skill.language === lang)
             )).map((lang, index) => (
                  <MenuItem key={index} value={lang}>
                    {lang}
                  </MenuItem>
             ))}
```

Reply:
Thx for the tips, never done this before, therefore code is not that clean, but it works.
<img width="1042" alt="Screenshot 2020-07-24 at 08 36 42" src="https://user-images.githubusercontent.com/22850021/88391886-08331300-cd89-11ea-8a96-42f8c08af82f.png">
<img width="496" alt="Screenshot 2020-07-24 at 08 34 43" src="https://user-images.githubusercontent.com/22850021/88391890-0a956d00-cd89-11ea-99e2-fef830f666eb.png">
Also it looks responsive to me. Will wait for another review if it looks good

Reply:
That looks good. The only edit that can possibly be made is having the delete button stay on the left and shrinking the field to fit that there. We can discuss this with the team.

Reply:
I like how this remove button visually separates languages one from each other on mobile screens. But if everyone else wants it to be on the left, no problem, I'll rewrite later.

Reply:
Might be better separating with a margin instead, but we can patch that later.